python/datasources/cdc_miovd.py

```diff
@@ -0,0 +1,139 @@
+1. Open your web browser and navigate to the CDC data portal:
+    - (https://data.cdc.gov/Injury-Violence/Mapping-Injury-Overdose-and-Violence-County/psx4-wq38/data_preview).
+2. Select "Actions" and choose "Query Data."
+4. Filter the dataset by intent to display either "FA_Homicide" for homicide reports
+   or "FA_Suicide" for suicide reports.
+
+Note: If the provided link does not work, please refer to the URL in the comment section below.
+
+URL for manual access:
+https://data.cdc.gov/Injury-Violence/Mapping-Injury-Overdose-and-Violence-County/psx4-wq38/about_data
+
+Last Updated: 6/3/2025
+"""
+
+import pandas as pd
+
+from datasources.data_source import DataSource
+from ingestion import dataset_utils, gcs_to_bq_util, merge_utils
+from ingestion import standardized_columns as std_col
+from ingestion.constants import CURRENT, HISTORICAL
+
+
+class CDCMIOVDData(DataSource):
+    # MIOVD constants
+    CONDITIONS = ["gun_violence_homicide", "gun_violence_suicide"]
+    DIRECTORY = "cdc_miovd"
+
+    FILE_NAME_MAP = {
+        "gun_violence_homicide": "gun_homicides-county-all.csv",
+        "gun_violence_suicide": "gun_suicides-county-all.csv",
+    }
+
+    # CSV parsing constants
+    CSV_COLS_TO_USE = {"Period", "ST_NAME", "NAME", "Count", "Rate", "TTM_Date_Range", "GEOID"}
+    DTYPE = {"Period": str, "GEOID": str}
+    NA_VALUES = ["1-9", "-999.0", "10-50"]
+
+    # MIOVD column names to HET standardized column names
+    CSV_TO_STANDARD_COLS = {
+        "Period": std_col.TIME_PERIOD_COL,
+        "NAME": std_col.COUNTY_NAME_COL,
+        "ST_NAME": std_col.STATE_NAME_COL,
+        "GEOID": std_col.COUNTY_FIPS_COL,
+    }
+
+    # Columns to merge the homicide and suicide dataframes
+    MERGE_COLS = [
+        std_col.TIME_PERIOD_COL,
+        std_col.COUNTY_NAME_COL,
+        std_col.STATE_NAME_COL,
+        std_col.COUNTY_FIPS_COL,
+        "is_ttm",
+    ]
+
+    @staticmethod
+    def get_id():
+        return "CDC_MIOVD_DATA"
+
+    @staticmethod
+    def get_table_name():
+        return "cdc_miovd_data"
+
+    def upload_to_gcs(self, gcs_bucket, **attrs):
+        raise NotImplementedError("upload_to_gcs should not be called for CDCMIOVD")
+
+    def write_to_bq(self, dataset, gcs_bucket, **attrs):
+        demo_type = self.get_attr(attrs, "demographic")
+        geo_level = self.get_attr(attrs, "geographic")
+
+        # Load data for both conditions
+        homicides_df = self.load_condition_data("gun_violence_homicide")
+        suicides_df = self.load_condition_data("gun_violence_suicide")
+
+        # Merge homicide and suicide data
+        df = merge_utils.merge_dfs_list([homicides_df, suicides_df], self.MERGE_COLS)
+        df = merge_utils.merge_state_ids(df)
+        df = merge_utils.merge_county_names(df)
+
+        # Split into annual and TTM dataframes using the is_ttm flag
+        annual_df = df.copy().drop(columns=["is_ttm"])
+        ttm_df = df[df["is_ttm"]].copy().drop(columns=["is_ttm"])

```

Comment:
i think this might be wrong; we need to ensure that the TTM rows are filtered out from the annual df as well, otherwise there will be additional entries in our historical calculations etc. Right now the annual_df still includes the TTM rows, you're only dropping the ttm boolean column. i think we need something like

```
annual_df = df[~df["is_ttm"]].copy().drop(columns=["is_ttm"])  # Only non-TTM rows
ttm_df = df[df["is_ttm"]].copy().drop(columns=["is_ttm"])      # Only TTM rows
```